pkg/actions/lua/lakefs/client.go

```diff
@@ -45,6 +46,9 @@ func newLakeFSRequest(ctx context.Context, user *model.User, method, reqURL stri
 		body = bytes.NewReader(data)
 	}
 
+	// Set no_hooks to true to disable recursive hooks
+	ctx = context.WithValue(ctx, graveler.ContextKeyNoHooks, true)
```

Comment:
I get why this works but I'm really not a fan of this approach, especially because it's implicit. 
Think of user doing commit to **another** branch they don't need this skip...